sources/bloodhound/src/lib.rs

```
@@ -0,0 +1,130 @@
+use std::{ffi::OsStr, fs::File, io::BufRead, io::BufReader, process::Command};
+
+pub mod args;
+pub mod output;
+pub mod results;
+
+/// Reads a file and checks if the given `search_str` is present in its contents.
+pub fn look_for_string_in_file(path: &str, search_str: &str) -> Option<bool> {
+    if let Ok(modules) = File::open(path) {
```

It is unusual to swallow the potential I/O error here, and the potential execution error in `look_for_string_in_output`, which is part of why this function reads a bit "off" - it doesn't benefit from `?` or more condensed syntax like:
```
let reader = BufReader::new(File::open(path)?;
reader.lines().any(|line| line == search_str)
```

It's also prone to misuse or confusion. A check may interpret a `None` result as "I didn't find this problematic string in the config file" and report success, when a skip result might be more appropriate ("I couldn't run this check because the file didn't exist").

I would recommend returning `Result<Option<bool>>` and just dealing with the three cases (error / some / none) in the caller.